### PR TITLE
add patch to h5py 2.8.0 easyconfigs to avoid MPI_Init on 'import h5py'

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
node3129.skitty.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/02cb47266bcc6cf53b2d2d8c26544e1d for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
node2492.golett.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/f606580381bcd31455565c23e88c4775 for a full test report.

File: easybuild/easyconfigs/h/h5py/h5py-2.8.0-foss-2018b-Python-2.7.15.eb [hunk 1]
```diff
@@ -14,20 +14,22 @@ toolchainopts = {'usempi': True}
...
     ('Python', '2.7.15'),
     ('HDF5', '1.10.2'),
     ('pkgconfig', '1.3.1', '-Python-%(pyver)s'),
 ]
 
-sanity_check_paths = {
-    'files': [],
-    'dirs': ['lib/python%(pyshortver)s/site-packages/'],
-}
+use_pip = False
```
Comment (Flamefire) [inline easybuild/easyconfigs/h/h5py/h5py-2.8.0-foss-2018b-Python-2.7.15.eb:29]: Would you like to rather use pip with https://github.com/easybuilders/easybuild-easyconfigs/pull/11044 ?
I'd recommend doing that and also adding the `sanity_pip_check = True`

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
node3105.skitty.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/2d71f48d6e4d3df4811f4698c3bdc053 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 18 out of 18 (4 easyconfigs in this PR)
easybuild.novalocal - Linux centos linux 7.8.2003, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 2.7.5
See https://gist.github.com/5d59419853fec8e6d29ba59c6289c4f0 for a full test report.

**Comment by verdurin** [comment]
@boegelbot please test @ generoso

**Comment by verdurin** [comment]
Strange error on `generoso`:
```
checking size of size_t... configure: error: in `/tmp/boegelbot/Python/3.6.6/intel-2018b/Python-3.6.6':
configure: error: cannot compute sizeof (size_t)
See `config.log' for more details
 (at easybuild/easybuild-framework/easybuild/tools/run.py:533 in parse_cmd_output)
```

**Comment by boegel** [comment]
@verdurin That's because the toolchains used here are too old for CentOS 8, it's simply not worth the effort to try and fix those problems imho...

**Comment by boegel** [comment]
@verdurin Only the `intel/2018b` toolchain is problematic on CentOS 8, so I submitted a test report for only the easyconfigs using `foss/2018b`

**Comment by verdurin** [comment]
@boegel is it worth noting that lack of compatibility with `CentOS 8` or do we assume anyone running that will install something newer?

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 7 out of 9 (7 easyconfigs in this PR)
node373.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/c2bb82aebe3f9265117ddddb65bbae0e for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 7 out of 9 (7 easyconfigs in this PR)
node109.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/182162665dd41cda64b0513252e88735 for a full test report.

**Comment by boegel** [comment]
> @boegel is it worth noting that lack of compatibility with `CentOS 8` or do we assume anyone running that will install something newer?

It's probably worth documenting that somewhere, but does that imply we need to start keeping track of a compatibility matrix between different Linux distro versions and toolchains?

We can open an issue mentioning the error message, like I did for `GCCcore` 6.4.0 in #10950, so people will hit it when searching for the error?

**Comment by boegel** [comment]
@lexming It looks like your `Python/2.7.15-intel-2018b` installation is missing `cffi`?!

**Comment by lexming** [comment]
@boegel that's indeed the case, re-installing `Python/2.7.15-intel-2018b` and re-testing...

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
node321.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/7fd783c76dba0bef9526f6c6ffe24aef for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
node127.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/38a787708b24075c4e0a8b950207216f for a full test report.

**Review by lexming — APPROVED** [review]
LGTM

**Comment by lexming** [comment]
Since the issue with `intel/2018b` and `CentOS8` is not a regression caused by this PR, I'm going ahead and merging.

Going in, thanks @boegel !

**Pull request merged**